### PR TITLE
Migration: Update autodocs migration to deal with `docs.docsPage`

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: https://github.com/storybookjs/storybook/pull/20364#issuecomment-1362453379

## What I did

Updated the `docs.autodocs` migration to convert/copy over the old value and warn if you need to change tags.

## How to test

Run migration against some converted and unconverted projects.

### CODE REVIEW

File: code/lib/cli/src/automigrate/fixes/autodocs-true.ts [hunk 1]
```diff
@@ -32,11 +34,50 @@ export const autodocsTrue: Fix<AutodocsTrueFrameworkRunOptions> = {
...
+      };
+    }
+
     return docs?.autodocs === undefined ? { main } : null;
   },
 
-  prompt() {
-    const AutodocsTrueFormatted = chalk.cyan(`docs: { autodocs: true }`);
+  prompt({ value }) {
+    const AutodocsTrueFormatted = chalk.cyan(
```
Comment (shilman) [inline code/lib/cli/src/automigrate/fixes/autodocs-true.ts:56]:
```suggestion
    const autodocsFormatted = chalk.cyan(
```

File: code/lib/cli/src/automigrate/fixes/autodocs-true.ts [hunk 2]
```diff
@@ -32,11 +34,50 @@ export const autodocsTrue: Fix<AutodocsTrueFrameworkRunOptions> = {
...
+    if (value) {
+      return dedent`
+      We've changed the configuration of autodocs (previous docsPage), so now the value:
+        - docs.autodocs: true -- means automatically create docs for every CSF file
+        - docs.autodocs: 'tag' -- means only create autodocs for CSF files with the 'autodocs' tag
+        - docs.autodocs: false -- means never create autodocs
+
+      Based on your prior configuration,  we can set the \`docs.autodocs\` to keep your old behaviour:
+
+      ${AutodocsTrueFormatted}
```
Comment (shilman) [inline code/lib/cli/src/automigrate/fixes/autodocs-true.ts:69]:
```suggestion
      ${autodocsFormatted}
```

File: code/lib/cli/src/automigrate/fixes/autodocs-true.ts [hunk 3]
```diff
@@ -51,10 +92,11 @@ export const autodocsTrue: Fix<AutodocsTrueFrameworkRunOptions> = {
     `;
   },
 
-  async run({ result: { main }, dryRun }) {
+  async run({ result: { main, value }, dryRun }) {
     logger.info(`✅ Setting 'docs.autodocs' to true in main.js`);
     if (!dryRun) {
-      main.setFieldValue(['docs', 'autodocs'], true);
+      main.removeField(['docs', 'docsPage']);
+      main.setFieldValue(['docs', 'autodocs'], value ?? true);
```
Comment (shilman) [inline code/lib/cli/src/automigrate/fixes/autodocs-true.ts:99]: If value is undefined do we want it to default to `true` or to `'tag'`?

**Review by shilman — APPROVED** [review]
LGTM but looks like there are some TS issues @tmeasday

File: code/lib/cli/src/automigrate/fixes/autodocs-true.ts (hunk 3 above)
Comment (tmeasday) [reply code/lib/cli/src/automigrate/fixes/autodocs-true.ts:99]: To `true`. That means they didn't previously have a value, which means they are upgrading, which means we want to set the behaviour to completely automatic.

**Review by shilman — COMMENTED** [review]
@tmeasday can you fix the TS error here?

**Pull request merged**